hibernate-reactive-core/src/test/java/org/hibernate/reactive/NonNullableManyToOneTest.java

```diff
@@ -0,0 +1,123 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+
+import org.hibernate.cfg.Configuration;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vertx.ext.unit.TestContext;
+
+public class NonNullableManyToOneTest extends BaseReactiveTest {
+
+	@Override
+	protected Configuration constructConfiguration() {
+		Configuration configuration = super.constructConfiguration();
+		configuration.addAnnotatedClass( Artist.class );
+		configuration.addAnnotatedClass( Painting.class );
+		return configuration;
+	}
+
+	@Before
+	public void populateDB(TestContext context) {
+		Artist artemisia = new Artist( "Grand Master Painter" );
+		artemisia.id = 1L;
+		Painting painting = new Painting( "Mona Lisa");
+		painting.id = 2L;
+		artemisia.addPainting( painting );
+
+		test( context, getMutinySessionFactory()
+				.withTransaction( s -> s.persistAll( painting, artemisia ) ) );
+	}
+
+	@After
+	public void cleanDB(TestContext context) {
+		test( context, deleteEntities( "Painting", "Artist" ) );
+	}
+
+	@Test
+	public void testNonNullableSuccess(TestContext context) {
+		test( context,
+				openMutinySession().chain( session -> session
+								.createQuery( "from Artist", Artist.class )
+								.getSingleResult() )
+						// We are checking `.getPaintings()` to verify success in persistence
+						.onItem().invoke( Artist::getPaintings )
+		);
+	}
+
+	@Entity(name = "Painting")
+	@Table(name = "painting")
+	public static class Painting {
+		@Id
+		public Long id;
+		String name;
+
+		@JoinColumn(name = "author_id", nullable = false)
+		@ManyToOne(fetch = FetchType.LAZY, targetEntity = Artist.class) //, optional = false)
```

Comment:
Maybe you could add another association that has `optional = false`. This way we are sure that both settings are working

Reply:
fixed